modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbStorage.java

```diff
@@ -0,0 +1,221 @@
+        try {
+            byte[] keyBytes = key.keyBytes();
+
+            return new SimpleDataRow(keyBytes, db.get(keyBytes));
+        }
+        catch (RocksDBException e) {
+            throw new StorageException("Failed to read data from the storage", e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public synchronized void write(DataRow row) throws StorageException {
+        try {
+            db.put(row.keyBytes(), row.valueBytes());
+        }
+        catch (RocksDBException e) {
+            throw new StorageException("Filed to write data to the storage", e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public synchronized void remove(SearchRow key) throws StorageException {
+        try {
+            db.delete(key.keyBytes());
+        }
+        catch (RocksDBException e) {
+            throw new StorageException("Failed to remove data from the storage", e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public synchronized void invoke(SearchRow key, InvokeClosure clo) throws StorageException {
+        try {
+            byte[] keyBytes = key.keyBytes();
+            byte[] existingDataBytes = db.get(keyBytes);
+
+            clo.call(new SimpleDataRow(keyBytes, existingDataBytes));
+
+            switch (clo.operationType()) {
+                case WRITE:
+                    db.put(keyBytes, clo.newRow().valueBytes());
+
+                    break;
+
+                case REMOVE:
+                    db.delete(keyBytes);
+
+                    break;
+
+                case NOOP:
+                    break;
+            }
+        }
+        catch (RocksDBException e) {
+            throw new StorageException("Failed to access data in the storage", e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public Cursor<DataRow> scan(Predicate<SearchRow> filter) throws StorageException {
+        return new ScanCursor(db.newIterator(), filter);
+    }
+
+    /** {@inheritDoc} */
+    @Override public void close() {
+        if (db != null)
+            db.close();
+
+        if (options != null)
+            options.close();
+
+        if (comparator != null)
+            comparator.close();
+
+        if (comparatorOptions != null)
+            comparatorOptions.close();
+    }
+
+    private static class ScanCursor implements Cursor<DataRow> {
+        private final RocksIterator iter;
+
+        private final Predicate<SearchRow> filter;
+
+        private ScanCursor(RocksIterator iter, Predicate<SearchRow> filter) {
+            this.iter = iter;
+            this.filter = filter;
+
+            iter.seekToFirst();
+
+            hasNext();
+        }
+
+        /** {@inheritDoc} */
+        @NotNull @Override public Iterator<DataRow> iterator() {
+            return this;
+        }
+
+        /** {@inheritDoc} */
+        @Override public boolean hasNext() {
+            while (iter.isValid() && !filter.test(new SimpleDataRow(iter.key(), null)))
```

Comment:
you should also call the `status` method in case `isValid` returns `false`, see https://github.com/facebook/rocksdb/wiki/Iterator#error-handling